sys/riscv/riscv/cheri_revoke_machdep.c

```diff
@@ -141,7 +141,7 @@ vm_do_cheri_revoke(int *res, const struct vm_cheri_revoke_cookie *crc,
 			"beq x0, %[ok], 1f\n\t"
 			"sc.c.cap %[cutr], (%[cutp])\n\t"
 			"1:\n\t"
-		  : [ok] "=r" (ok), [cscratch] "=&C" (cscratch),
+		  : [ok] "=&r" (ok), [cscratch] "=&C" (cscratch),
```

Comment:
Shouldn't this be `+&r` as well?

Reply:
No, the value prior to the asm statement isn’t read (and isn’t initialised)

Reply:
The difference is Morello can use condition codes to compare whereas RISC-V has to use a CSEQX to get the bool

Reply:
I see now, thank you.